lib/index.d.ts

```diff
@@ -228,14 +259,52 @@ export class Component<
    * Fetches a value from the component's configuration map (a combination of
    * values supplied in the config() getter and defaults applied automatically).
    */
-  getConfig<K extends keyof ConfigOptions<StateT, AppStateT, ContextRegistryT>>(key: K): this['config'][K];
+  getConfig<K extends keyof ConfigOptions<StateT, AppStateT, ContextRegistryT, ParamT>>(key: K): this['config'][K];
 
   /** Sets a value in the component's configuration map after element initialization */
-  setConfig<K extends keyof ConfigOptions<StateT, AppStateT, ContextRegistryT>>(
+  setConfig<K extends keyof ConfigOptions<StateT, AppStateT, ContextRegistryT, ParamT>>(
     key: K,
     val: ConfigOptions<StateT, AppStateT, ContextRegistryT>[K],
   ): void;
 
+  /**
+   * set the params for the this component
+   * triggers a component update
+   * if shouldComponentUpdate callback returns true
+   */
+  setParams(params: Partial<ParamT>): void;
+
+  /**
+   * Same API as react's `shouldComponentUpdate` usage
+   * if child component implements this method, parent implmentation wil be discarded
+   * only difference is the `params` or `state` could sometimes be null indicating that
+   * the update is not related to `params` or `state`
+   *
+   * To be overridden by subclasses, defining conditional logic for whether
+   * a component should rerender its template given the state and params to be applied.
+   * In most cases this method can be left untouched, but can provide improved
+   * performance when dealing with very many DOM elements.
+   *
+   * @example
+   * shouldComponentUpdate(params, state) {
+   *   if (params.bookmark.id !== this.params.bookmark.id) {
+   *     return false;
```

Comment:
In this example, shouldn't it `return true` in the case where the params update is changing it to a totally different bookmark?